sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -370,9 +378,56 @@ static ActivityExtensions()
         private static Func<object, bool>? ActivitySourceHasListenersMethod;
         private static Func<string, string?, ICollection<KeyValuePair<string, object>>?, object?>? CreateActivityLinkMethod;
         private static Func<ICollection<KeyValuePair<string,object>>?>? CreateTagsCollectionMethod;
-
+        private static Func<Activity, string, object?>? GetCustomPropertyMethod;
+        private static Action<Activity, string, object>? SetCustomPropertyMethod;
+        private static string AzureSdkScopeLabel = "az.sdk.scope";
```

Comment:
From what I understand, `ActivityExtensions` type is a hack that uses reflection to access methods that we can't bind in compile time. `"az.sdk.scope"` is specific for nested scope only.